src/zjs_util.h

```diff
@@ -15,6 +15,16 @@
 #include "zjs_common.h"
 #include "zjs_error.h"
 
+#ifdef ZJS_TRACE_MALLOC
+typedef struct mem_stats {
+    void *ptr;
+    char *file;
+    char *func;
+    int line;
+    struct mem_stats *next;
```

Comment:
Looks like you started out with this as a list and later changed to an array, probably so you wouldn't need dynamic allocations? So you can drop the `next` field now and better to rename `mem_list` to `mem_array`, `mem_in_use`,  `mem_tracker`, or something.